PaddleAudio/paddleaudio/models/wav2vec2/models.py

```diff
@@ -0,0 +1,1028 @@
+        hidden_states = hidden_states + self.feed_forward(hidden_states)
+        hidden_states = self.final_layer_norm(hidden_states)
+
+        outputs = (hidden_states, )
+
+        if output_attentions:
+            outputs += (attn_weights, )
+
+        return outputs
+
+
+class Wav2Vec2Encoder(nn.Layer):
+    def __init__(self, config):
+        super().__init__()
+        self.config = config
+        self.pos_conv_embed = Wav2Vec2PositionalConvEmbedding(config)
+        self.layer_norm = nn.LayerNorm(config.hidden_size,
+                                       epsilon=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout)
+        self.layers = nn.LayerList([
+            Wav2Vec2EncoderLayer(config)
+            for _ in range(config.num_hidden_layers)
+        ])
+
+    def forward(
+        self,
+        hidden_states,
+        attention_mask=None,
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=True,
+    ):
+
+        assert output_hidden_states is False
+        assert output_attentions is False
+        # all_hidden_states = () if output_hidden_states else None
+        all_self_attentions = () if output_attentions else None
+
+        if attention_mask is not None:
+            # make sure padded tokens output 0
+            hidden_states[~attention_mask] = 0.0
+
+            # extend attention_mask
+            attention_mask = (1.0 - attention_mask[:, None, None, :].to(
+                dtype=hidden_states.dtype)) * -10000.0
+            attention_mask = attention_mask.expand(attention_mask.shape[0], 1,
+                                                   attention_mask.shape[-1],
+                                                   attention_mask.shape[-1])
+        position_embeddings = self.pos_conv_embed(hidden_states)
+        hidden_states = hidden_states + position_embeddings
+        hidden_states = self.layer_norm(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+
+        for layer in self.layers:
+            #  if output_hidden_states:
+            #all_hidden_states = all_hidden_states + (hidden_states,)
+
+            # add LayerDrop (see https://arxiv.org/abs/1909.11556 for description)
+            dropout_probability = np.random.uniform(0, 1)
+
+            layer_outputs = layer(hidden_states,
+                                  attention_mask=attention_mask,
+                                  output_attentions=output_attentions)
+            hidden_states = layer_outputs[0]
+
+        #  if output_attentions:
+        #  all_self_attentions = all_self_attentions + (layer_outputs[1],)
+
+    #   if not return_dict:
+    #return tuple(v for v in [hidden_states, all_hidden_states, all_self_attentions] if v is not None)
+
+        return BaseModelOutput(
+            last_hidden_state=hidden_states,
+            hidden_states=None,
+            attentions=None,
+        )
+
+
+class Wav2Vec2PreTrainedModel(nn.Layer):
```

Comment:
It would be better to rename `PreTrained` -> `Pretrained`

Reply:
fixed